File: src/main/java/club/someoneice/thursday_delight/chili/Chili.java
package club.someoneice.thursday_delight.chili;

import club.someoneice.thursday_delight.BlockList;
import club.someoneice.thursday_delight.ThursdayMain;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.food.FoodProperties;
import net.minecraft.world.item.BlockItem;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;

public class Chili extends BlockItem {
    public Chili() {
        super(BlockList.ChiliCrop.get() ,new Properties().tab(ThursdayMain.TAB).food(new FoodProperties.Builder().nutrition(2).saturationMod(0.1f).fast().alwaysEat().build()).fireResistant());
    }

    @Override
    public ItemStack finishUsingItem(ItemStack itemStack, Level world, LivingEntity player) {
        player.setRemainingFireTicks(20 * 3);
        return super.finishUsingItem(itemStack, world, player);
    }
}

File: src/main/kotlin/club/someoneice/thursday_delight/chili/Chili.kt
package club.someoneice.thursday_delight.chili

import club.someoneice.thursday_delight.RegisterInit
import club.someoneice.thursday_delight.ThursdayMain
import net.minecraft.entity.LivingEntity
import net.minecraft.item.BlockItem
import net.minecraft.item.FoodComponent
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class Chili: BlockItem(RegisterInit.ChiliCrop ,Settings().group(ThursdayMain.thursdayGroup).food(FoodComponent.Builder().hunger(2).saturationModifier(0.1f).snack().alwaysEdible().build()).fireproof()) {
    override fun finishUsing(stack: ItemStack, world: World, user: LivingEntity): ItemStack {
        user.fireTicks = 20 * 3
        return super.finishUsing(stack, world, user)
    }
}